fairlearn/metrics/_extra_metrics.py

```diff
@@ -6,38 +6,63 @@
 These are metrics which are not part of `scikit-learn`.
 """
 
+import numpy as np
 import sklearn.metrics as skm
 
 from ._balanced_root_mean_squared_error import _balanced_root_mean_squared_error  # noqa: F401
 from ._mean_predictions import mean_prediction, _mean_overprediction, _mean_underprediction  # noqa: F401,E501
 from ._selection_rate import selection_rate  # noqa: F401,E501
 
+_NEED_TWO_UNIQUE_Y_VALS = "Must have two unique y values"
+_NEED_POS_LABEL_IN_Y_VALS = "Must have pos_label in y values"
 
-def true_positive_rate(y_true, y_pred, sample_weight=None):
+
+def _get_labels_for_confusion_matrix(y_true, y_pred, pos_label):
+    """Figure out the labels argument for skm.confusion_matrix.
+
+    This assumes the input arrays are binary, and we need to specify a pos_label.
+    Given the way the calling routines work, this is achieved by ensuring that
+    the pos_label is last in the (two element) list.
+    """
+    my_labels = list(np.unique(np.concatenate((y_true, y_pred), axis=None)))
+    if len(my_labels) != 2:
```

Comment:
Right now we require it to be 0 and 1 because some of the moments depend on this. This seems somewhat more general, so I want to be careful and not break that. This code looks like it could allow any two values to pass through, but perhaps I'm missing something

Reply:
Moments call into metrics? I hadn't realised that. But in any case, this doesn't change the existing behaviour.

Reply:
The current behavior had 0/1 explicitly from what I can tell.

Reply:
The current behavior does explicitly state `labels=[0,1]` when calling `confusion_matrix()`, but scikit-learn does not raise an error if you do not adhere to 0/1, it will just ignore other values. I.e. `confusion_matrix(y_true=[0,2,2,0], y_pred=[0,2,0,0], labels=[0,1])` returns `array([[2, 0], [0, 0]])`. So if you want to *require* 0/1, we need to add an additional check.